app/src/main/java/de/tum/in/tumcampusapp/adapters/ExamListAdapter.java

```diff
@@ -90,6 +79,50 @@ public View getView(int position, View convertView, ViewGroup parent) {
         return view;
     }
 
+    private void setGradeBackground(ViewHolder holder, String grade){
+        //if(holder.gradeBackground instanceof ShapeDrawable){
+            //Drawable bg = (ShapeDrawable) holder.gradeBackground;
+            //Resources res = holder.tvGrade.getResources();
+            TextView view = holder.tvGrade;
+
+            // TODO find elegant/better solution that only needs one file
+            switch (grade){
+                case "1,0": view.setBackgroundResource(R.drawable.grade_1_0);
+                    break;
+                case "1,3":
+                case "1,4": view.setBackgroundResource(R.drawable.grade_1_3);
```

Comment:
map the grade to an ressource id separately and then call `view.setBackgroundResource()` once